File: Qualification_Round/Nesting_Depth.cpp
/* author           : Akash kumar 
   codejam username : Akash@1996
*/


#include<bits/stdc++.h>
using namespace std;

#define ll long long int
#define mod 1000000007
#define pb push_back

void solve(string s)
{
   int n=s.size();
   int c=0;
   for(int i=0;i<n;++i)	
   {
   	if(s[i]=='0')
   	{
   		if(c!=0)
   		{
   			cout<<")";
   		}
   		cout<<s[i];
   		c=0;
   	}
   	else
   	{
   		if(c==0)
		{
			cout<<"("<<s[i];
			c++;
		}
		else
		{
			cout<<s[i];
		}
   	}
   }
   if(s[n-1]=='1')
   {
	   cout<<")";
   }
   cout<<"\n";
}

int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    int t;
    cin>>t;
    for(int i=1;i<=t;++i)
    {
    	string s;
    	cin>>s;
    	cout<<"Case #"<<i<<": ";
    	solve(s);
    }
	return 0;
}

File: Qualification_Round/Nesting_Depth.py
"""
author            :   Akash kumar
codejam username  :   Akash@1996
"""

T=1
for _ in range(int(input())):
    s=str(input())
    c=0
    print("Case #"+str(T)+": ",end="")
    for i in range(len(s)): 
        if s[i]=='0':
           if c!=0:
             print(")",end="")
           print(s[i],end="")
           c=0
        else:
           if c==0:
             print("(",end="")
             print(s[i],end="")
             c+=1
           else:
              print(s[i],end="")
    if s[len(s)-1]=='1':
        print(")",end="")
    T+=1
    print()
         
               
